### PR TITLE
For cache/map split brain tests, partitions should be assigned at the beginning

### LINKED ISSUE #7042: CacheSplitBrainTest.testPassThroughCacheMergePolicy
```
java.lang.IllegalStateException: Partitions can't be assigned since cluster-state: FROZEN
    at com.hazelcast.spi.impl.operationservice.impl.Invocation.notifyWithExceptionWhenTargetIsNull(Invocation.java:341)
    at com.hazelcast.spi.impl.operationservice.impl.Invocation.initInvocationTarget(Invocation.java:304)
    at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:234)
    at com.hazelcast.spi.impl.operationservice.impl.Invocation.invokeInternal(Invocation.java:210)
    at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:180)
    at com.hazelcast.spi.impl.operationservice.impl.OperationServiceImpl.invokeOnPartition(OperationServiceImpl.java:314)
    at com.hazelcast.cache.impl.AbstractInternalCacheProxy.invoke(AbstractInternalCacheProxy.java:113)
    at com.hazelcast.cache.impl.AbstractInternalCacheProxy.invoke(AbstractInternalCacheProxy.java:134)
    at com.hazelcast.cache.impl.AbstractInternalCacheProxy.putAsyncInternal(AbstractInternalCacheProxy.java:193)
    at com.hazelcast.cache.impl.AbstractCacheProxy.put(AbstractCacheProxy.java:200)
    at com.hazelcast.cache.impl.CacheProxy.put(CacheProxy.java:75)
    at com.hazelcast.cache.impl.CacheProxy.put(CacheProxy.java:135)
    at com.hazelcast.cache.merge.CacheSplitBrainTest.testPassThroughCacheMergePolicy(CacheSplitBrainTest.java:238)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-nightly/com.hazelcast$hazelcast/796/testReport/junit/com.hazelcast.cache.merge/CacheSplitBrainTest/testPassThroughCacheMergePolicy/

### PR DESCRIPTION
**Pull request by serkan-ozal**
… and we should ensure that merging is not applied before mutations are done on cache/map data structure

Fixes #7042

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/map/MergePolicyTest.java [hunk 1]
```diff
@@ -233,15 +270,23 @@ private Config newConfig(String mergePolicy, String mapName) {
...
         @Override
         public void stateChanged(LifecycleEvent event) {
-            if (event.getState() == LifecycleEvent.LifecycleState.MERGED) {
+            if (event.getState() == LifecycleEvent.LifecycleState.MERGING) {
+                try {
+                    mergeBlockingLatch.await(30, TimeUnit.SECONDS);
+                } catch (InterruptedException e) {
+                    ExceptionUtil.rethrow(e);
+                }
+            } else if (event.getState() == LifecycleEvent.LifecycleState.MERGED) {
```
Comment (jerrinot) [inline hazelcast/src/test/java/com/hazelcast/map/MergePolicyTest.java]: do we need this branch at all? 
if the merging process is blocked then the only way to unblock it is to use the latch, isn't it?

Comment (serkan-ozal) [inline hazelcast/src/test/java/com/hazelcast/map/MergePolicyTest.java]: in here, we ensure that merging is finished

**Comment by serkan-ozal** [comment]
@jerrinot do you have any other comment on this PR?

File: hazelcast/src/test/java/com/hazelcast/map/MergePolicyTest.java (hunk 1 above)
Comment (jerrinot) [inline hazelcast/src/test/java/com/hazelcast/map/MergePolicyTest.java]: I understand it now

**Comment by jerrinot** [comment]
:+1:

**Pull request merged**